internal/controllers/gateway/gateway_utils_test.go

```diff
@@ -130,15 +130,25 @@ func TestGetListenerStatus_no_duplicated_Detached_condition(t *testing.T) {
 	require.NoError(t, err)
 	require.Len(t, statuses, 1, "only one listener status expected as only one listener was defined")
 	listenerStatus := statuses[0]
-	assertOnlyOneConditionOfType(t, listenerStatus.Conditions, gatewayv1beta1.ListenerConditionAccepted)
+	assertOnlyOneConditionOfTypes(t, listenerStatus.Conditions,
+		gatewayv1beta1.ListenerConditionAccepted,
+		gatewayv1beta1.ListenerConditionReady,
+		gatewayv1beta1.ListenerConditionProgrammed,
+		gatewayv1beta1.ListenerConditionResolvedRefs,
+		gatewayv1beta1.ListenerConditionConflicted,
+	)
 }
 
-func assertOnlyOneConditionOfType(t *testing.T, conditions []metav1.Condition, typ gatewayv1beta1.ListenerConditionType) {
-	conditionNum := 0
-	for _, condition := range conditions {
-		if condition.Type == string(typ) {
-			conditionNum++
+func assertOnlyOneConditionOfTypes(t *testing.T, conditions []metav1.Condition, types ...gatewayv1beta1.ListenerConditionType) {
```

Comment:
Take it or leave it hint: you could consider using [`lo.CountValues()`](https://github.com/samber/lo#countvalues) or [`lo.CountValuesBy()`](https://github.com/samber/lo#countvaluesby) for this and compare the result with the number of entries in `conditions.`

Reply:
Thanks for this suggestion! I used `lo.CountValuesBy()` and the code is way simpler! PTAL